tests/integration-tests/tests/storage/kms_key_factory.py

```diff
@@ -0,0 +1,207 @@
+
+                # attach ParallelClusterPolicy to the role
+                logging.info("Attaching iam policy to the existing role {0}...".format(iam_role_name))
+                policy_attach_res = self.iam_client.attach_role_policy(
+                    RoleName=iam_role_name, PolicyArn=self.iam_policy_arn
+                )
+                logging.info("Iam role is ready: {0}".format(iam_role_name))
+                return iam_role_name
+
+        # if the role doesn't exist, create the iam role
+        logging.info("creating iam role {0} for creating KMS key...".format(iam_role_name))
+        # create iam role, add EC2 as trust entity
+        trust_relationship_policy_ec2 = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {"Effect": "Allow", "Principal": {"Service": "ec2.amazonaws.com"}, "Action": "sts:AssumeRole"}
+            ],
+        }
+        role_res = self.iam_client.create_role(
+            RoleName=iam_role_name,
+            AssumeRolePolicyDocument=json.dumps(trust_relationship_policy_ec2),
+            Description="Role for create custom KMS key",
+        )
+        policy_arn = self._create_iam_policies(iam_policy_name, scheduler)
+        # attach ParallelClusterPolicy to the role
+        logging.info("Attaching iam policy to the role {0}...".format(iam_role_name))
+        policy_attach_res = self.iam_client.attach_role_policy(RoleName=iam_role_name, PolicyArn=policy_arn)
+        time.sleep(20)
+        logging.info("Iam role is ready: {0}".format(iam_role_name))
+        return iam_role_name
+
+    def _create_iam_policies(self, iam_policy_name, scheduler):
+        # create the iam policy
+        # for different scheduler, attach different instance policy
+        logging.info("Creating iam policy {0} for iam role...".format(iam_policy_name))
+        file_loader = FileSystemLoader(pkg_resources.resource_filename(__name__, "/../../resources"))
+        env = Environment(loader=file_loader, trim_blocks=True, lstrip_blocks=True)
+        policy_filename = (
+            "batch_instance_policy.json" if scheduler == "awsbatch" else "traditional_instance_policy.json"
+        )
+        parallel_cluster_instance_policy = env.get_template(policy_filename).render(
+            partition="*", region="*", account_id=self.account_id, cluster_bucket_name="parallelcluster-*"
+        )
+
+        policy_res = self.iam_client.create_policy(
+            PolicyName=iam_policy_name, PolicyDocument=parallel_cluster_instance_policy
+        )
+        policy_arn = policy_res["Policy"]["Arn"]
+        return policy_arn
+
+    def _create_kms_key(self, scheduler):
+        # create KMS key
+        self.kms_client = boto3.client("kms")
+        if scheduler == "awsbatch":
+            key_alias = "alias/Integration_test_KMS_key_batch_{0}".format(self.region)
+        else:
+            key_alias = "alias/Integration3_test_KMS_key_{0}".format(self.region)
```

Comment:
why the two different formats for `key_alias`?

Reply:
This is wrote by mistake, fixed